eventually-postgres/src/store.rs

```diff
@@ -121,11 +149,23 @@ impl EventStoreBuilder {
 ///
 /// [`EventStore`]: struct.EventStore.html
 /// [`EventStoreBuilder`]: struct.EventStoreBuilder.html
-pub struct EventStoreBuilderMigrated {
-    client: Arc<Client>,
+pub struct EventStoreBuilderMigrated<Tls>
+where
+    Tls: MakeTlsConnect<Socket> + Clone + Send + Sync + 'static,
+    <Tls as MakeTlsConnect<Socket>>::Stream: Send + Sync,
+    <Tls as MakeTlsConnect<Socket>>::TlsConnect: Send,
+    <<Tls as MakeTlsConnect<Socket>>::TlsConnect as TlsConnect<Socket>>::Future: Send,
```

Comment:
Do you need these clauses here, if they're specified again in the `impl<Tls> EventStoreBuilderMigrated<Tls>` block?

Reply:
Yeah, sadly it seems I do. I'm not too happy with the super long trait bounds list, I wish there was a way to create an alias for that.. But if I remove any right now, there are many complaints

Reply:
https://doc.rust-lang.org/unstable-book/language-features/trait-alias.html Seems to be at least on the way there (though not sure it even allows replacing all four lines)

Reply:
Oh yeah indeed, it seems the trait bound is defined on the manager structure itself: https://docs.rs/bb8-postgres/0.4.0/src/bb8_postgres/lib.rs.html#18-20